src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -1147,7 +1147,8 @@ int shim::p2pEnable(bool enable, bool force)
         mDev->sysfs_put("p2p", "p2p_enable", err, "0");
     }
     if (!err.empty()) {
-        throw std::runtime_error("P2P is not supported");
+        std::cerr << "ERROR: Config P2P failed : P2P is not supported" << std::endl;
+        throw xrt_core::error(std::errc::operation_canceled);
```

Comment:
The exception should be handled by caller.  In this case xclP2PEnable (which an implementation of C-API) does correctly catch the exception and send the exception message.   

Don't attempt to handle the exception here at this level only to throw another exception.  

If the exception message and or exception type is not as desired then change it on the original throw line.

I agree that you should throw xrt_core::error with an error code, as you do on line 1151, but add the message paylload and remove the std::cerr.

Reply:
Will do thanks for the tip